src/Symfony/Component/CssSelector/Node/MatchingNode.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\CssSelector\Node;
+
+/**
+ * Represents a "<selector>:is(<subSelectorList>)" node.
+ *
+ * This component is a port of the Python cssselect library,
+ * which is copyright Ian Bicking, @see https://github.com/SimonSapin/cssselect.
+ *
+ * @author Hubert Lenoir <lenoir.hubert@gmail.com>
+ *
+ * @internal
+ */
+class MatchingNode extends AbstractNode
+{
+    /**
+     * @param array<NodeInterface> $arguments
+     */
+    public function __construct(
+        private readonly NodeInterface $selector,
+        private readonly array $arguments = []
+    ) {
+    }
+
+    public function getSelector(): NodeInterface
```

Comment:
We can probably use public readonly properties as well.

Reply:
Indeed!

Reply:
Not sure about this, deprecation handling is a lot more complex isnt it? 

Do we use public properties in the Symfony codebase?

Reply:
I think it's fine on `@internal` classes.